pkg/controller/controller.go

```diff
@@ -531,6 +582,13 @@ func (p *csiProvisioner) ProvisionExt(options controller.ProvisionOptions) (*v1.
 		req.VolumeContentSource = volumeContentSource
 	}
 
+	if options.PVC.Spec.DataSource != nil && rc.clone {
+		err = p.setCloneFinalizer(options.PVC.Spec.DataSource.Name, options.PVC.Namespace)
```

Comment:
Is there reason you pass name/namespace instead of full PVC? `setCloneFinalizer` does not need to query API server at all.